sparse_strips/vello_hybrid/src/render/wgpu.rs

```diff
@@ -97,6 +101,55 @@ impl Renderer {
 
         self.scheduler.do_scene(&mut junk, scene)
     }
+
+    /// Upload an image to the atlas texture.
+    pub fn copy_texture_to_atlas(
+        &mut self,
+        encoder: &mut CommandEncoder,
+        texture: &Texture,
```

Comment:
This API necessitates the image be uploaded to a texture prior to rendering even if that image is only ever used in the Atlas.

I wonder whether instead of passing a `wgpu::Texture` directly, we could pass a trait that implements an ability to write to the atlas? 

Then we could implement that trait for `Pixmap` or `Texture` ourselves so that, for users with a pixmap, they can avoid 1 copy (and instead write directly into the atlas with `queue.write_texture` or similar). The pixmap might need to be Arc'd but that's likely cheaper than paying the extra copy and paying for more texture memory

Reply:
Thanks, Taj! That’s a great suggestion. I’ve implemented your idea as I understood it, and the API now feels more ergonomic, customizable, and performant.

Fixed.